packages/aws-cdk-lib/cx-api/lib/cloud-artifact.ts

```diff
@@ -110,12 +111,36 @@ export class CloudArtifact {
 
   private renderMessages() {
     const messages = new Array<SynthesisMessage>();
+    // messageId: scopes[]
+    const acks = new Map<string, string[]>();
 
+    // collect any acknowledgements
+    for (const [_id, metadata] of Object.entries(this.manifest.metadata || { })) {
+      for (const entry of metadata) {
+        if (entry.type === cxschema.ArtifactMetadataEntryType.ACKNOWLEDGE) {
+          const data = (entry.data as cxschema.AcknowledgementMetadataEntry);
+          acks.set(data.id, data.scopes);
+        }
```

Comment:
I think all of this work should probably happen at synth time in the CDK app.

Reply:
I started going down that path and ended up with this solution for a couple of reasons, the main one being that we create a new instance of `Annotations` every time we create a message. We currently use the metadata as the storage mechanism. In order to do it at synth time we would need to implement a new storage mechanism. This probably wouldn't be too bad (probably just another construct we getOrCreate).

I can explore that.